public/javascripts/store.js

```diff
@@ -0,0 +1,39 @@
+//vuex state store to share state between all components
+function makeStore() {
+  var leftPanelClass = 'cc-leftPanel-mobile-hide',
+      userInfoDefault = {
+        email: '',
+        imgSrc: 'https://s.ytimg.com/yts/img/avatar_720-vflYJnzBZ.png'
+      },
+      userInfo = userInfoDefault;
+  if (firebase.auth().currentUser) {
+    var user = firebase.auth().currentUser;
+    user.imgSrc = user.photoURL || 'https://s.ytimg.com/yts/img/avatar_720-vflYJnzBZ.png';
+    userInfo = user;
+  }
+  return {
+    state: {
+      leftPanelClass,
+      userInfo
+    },
+    mutations: {
+      toggleState: function(state, newState, property) {
+        state[property] = newState;
+      },
+      setUser: function(state, user) {
+        if (user) {
```

Comment:
if we are talking about the setUser function, it is passed the 'user' object on which Object.assign() can't be used according to MDN, as it is full of getters and setters, see console logged value of user object: http://puu.sh/q2hVe/e450286c0e.png